source/common/upstream/cluster_manager_impl.cc

```diff
@@ -268,7 +268,11 @@ ClusterManagerImpl::ClusterManagerImpl(
               ->create(),
           main_thread_dispatcher,
           *Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
-              "envoy.service.discovery.v2.AggregatedDiscoveryService.DeltaAggregatedResources"),
+              dyn_resources.ads_config().transport_api_version() ==
+                      envoy::config::core::v3::ApiVersion::V3
+                  ? "envoy.service.discovery.v3.AggregatedDiscoveryService.DeltaAggregatedResources"
+                  : "envoy.service.discovery.v2.AggregatedDiscoveryService."
+                    "DeltaAggregatedResources"),
```

Comment:
Would it be better to put this logic in type_to_endpoint.cc so that we have all of the version selection logic in one place? WDYT?

Reply:
Possibly, but ADS is an exception; there is no mapping from type to endpoint... we would probably want to rename that module.

Reply:
Up to you. I think it would be better to co-locate this type of selection logic. Maybe at least put in a TODO if you don't want to do it now?

Reply:
TODOs added, I think this depends on where we go with https://github.com/envoyproxy/envoy/issues/10650.